AnkiDroid/src/main/java/com/ichi2/async/CollectionTask.java

```diff
@@ -162,7 +166,11 @@ public static CollectionTask launchCollectionTask(TASK_TYPE type, TaskData... pa
      */
     public static CollectionTask launchCollectionTask(TASK_TYPE type, @Nullable Listener listener, TaskData... params) {
         // Start new task
-        CollectionTask newTask = new CollectionTask(type, listener, sLatestStartedInstance);
+        CollectionTask newTask;
+        synchronized(sLatestCreatedInstance) {
```

Comment:
If you mark a method synchronized (as you do in the following / final commit) you are synchronizing on the "monitor" of the object itself itself (either the instance if it's an instance method, or the class monitor if it's a static - those are separate!).

If you do a synchronized block on a specific object as you do here, then you synchronize on that monitor.

So you are not actually synchronizing the way you want - here it's on sLatestCreatedInstance (which is incorrect anyway as the object itself is changed, meaning different threads would be using different object monitors to synchronize on), and in the method it's on the "this" object.

What you want is I think this, so this is on the same monitor as the method:

```suggestion
        synchronized {
```

Reply:
Ahhh sorry @Arthur-Milchior my syntax wasn't perfect, didn't test that one. You do have to specify an object to synchronize on. Synchronized methods use the monitor of "this", so I think saying this is actually correct - but definitely do a jacocoUnitTestReport on it so that it compiles and runs first before sending it to Travis :-)

```
synchronized(this) {
```

Reply:
https://stackoverflow.com/a/13264751/9910298